lib/internal/modules/cjs/loader.js

```diff
@@ -824,12 +825,30 @@ Module.runMain = function() {
   Module._load(process.argv[1], null, true);
 };
 
-Module.createRequireFromPath = (filename) => {
+function createRequireFromPath(filename) {
   const m = new Module(filename);
   m.filename = filename;
   m.paths = Module._nodeModulePaths(path.dirname(filename));
   return makeRequireFunction(m);
-};
+}
+
+Module.createRequireFromPath = createRequireFromPath;
+
+function createRequire(filename) {
+  let path;
+  if (typeof filename === 'string' && filename.startsWith('.')) {
+    throw new ERR_INVALID_CREATE_REQUIRE_PATH(filename);
+  } else if (typeof filename === 'object' && filename instanceof URL ||
+    typeof filename === 'string' && !filename.startsWith('/')) {
```

Comment:
@guybedford as we now enforce absolute paths this logic should cause any string based URLs to go through `fileURLToPath` and fail accordingly

Reply:
This seems to align a lot better in concept to resolve the odd behaviour I was noting earlier — thanks 👍